### PR TITLE
712 - Show review accept/reject, commenting, and status tracking in sidebar

### LINKED ISSUE #712: Submit/accept/reject - Show whether submitted node has been submitted/accepted/rejected
See mockups below for the outline that shows up (either green, orange, or red) to signify the status of their submitted node.

Add a new section to the sidebar called "Review". This section will be used to show the user the status of their submitted node and any associated comments. They will also be able to comment back here.

When a user submits a node, the author would see that node on their end only and there would be an orange line around the nodes that have been submitted and not accepted/rejected yet.

Potentially: User should see this as well for nodes that were accepted/rejected

### PR DESCRIPTION
**Pull request by seanRong**
Closes #712 

Marks submitted, rejected nodes in a different color for both regular user and admin.

Adds a new section to the sidebar called "Review".  This shows node status as well as a comment thread for both the admin/user to comment on the node. 

* author will not be able to see rejected nodes after initial rejection. I feel this is correct, but is it intended for this ticket?

NEW as of Oct 26: scope change to have the color and acceptance modifications down in 680. This PR will only cover the comment system itself.

### CODE REVIEW

**Comment by yungalyx** [comment]
1. converting from draft to publish doesn't change the dotted outline
2. there's a millisecond where a second modal is shown right after i hit submit..

ezgif com-video-to-gif (1)

File: templates/vue/src/components/tapestry-node/ProgressBar.vue [hunk 1]
```diff
@@ -72,9 +72,11 @@ export default {
         case "draft":
           return "#999"
         case "submitted":
-          return "blue"
+          return "orange"
+        case "accept":
+          return "green"
         case "reject":
-          return "red"
+          return "#CC444B"
```
Comment (yungalyx) [inline templates/vue/src/components/tapestry-node/ProgressBar.vue:79]: create "red" constant for consistency uwu

**Comment by jaskwong** [comment]
@yungalyx  I think that's the case for users with adding privileges that are publishing the node instead of submitting it for review, I tested it and it works correctly for me when users without adding privileges go from private draft to "submit to administrators..." it has a yellow outline

Other issues:
1. when users with add privileges go from private draft to publish, the published node doesn't appear on the tapestry for other users (public or admin)

2. Not sure if this is intended: when users with add privileges go from private draft to publish, the new node has a grey outline around the node (but only the logged in user can see the node as noted in (1))

**Comment by seanRong** [comment]
Noting that fixes should probably be applied earlier on in this PR chain.

**Comment by jaskwong** [comment]
I'm not sure how to access the sidebar (I tried selecting the submitted, rejected, approved node and expanding the sidebar)
I'm also not sure how to add comments, (I tried adding the comments in the textbox and then approving the node, also tried with rejecting the node)v but then I can't find it again

Colour outlines work great!

**Comment by seanRong** [comment]
@jaskwong There should be an arrow on the right of your screen 
Screen Shot 2020-10-08 at 1 18 30 PM
Unless you're saying that when you click it it doesn't open?

**Comment by jaskwong** [comment]
@seanRong I can see the sidebar, in the description it states "Adds a new section to the sidebar called "Review". This shows node status as well as a comment thread for both the admin/user to comment on the node." I can't find the review section/comment thread

**Comment by seanRong** [comment]
That's odd will look into it

**Comment by jaskwong** [comment]
@seanRong Sorry I just found it! I just had to scroll down,  I think it's not very intuitive to find, is it possible to add an icon like (similar to the license/info icons) on the sidebar

**Comment by seanRong** [comment]
@jaskwong  still worrisome because I actually did make an icon as well (rip)

**Comment by jaskwong** [comment]
Some other issues:
- review section shows up on for all nodes, not just submitted ones 
- all users can read the comments from the admin in the review section, even after a node is published

**Comment by yungalyx** [comment]
@seanRong @narendrasss hey im not sure if im doing something wrong, or if this branch's implementations aren't showing up for me. I added nodes as a public user and i'm on admin side and I can't see any options for reject/accept nor can i see the comment thread.

i fixed the npm issue, I pulled latest, and I;m running on the correct branch.

712notworking

**Comment by seanRong** [comment]
You have to actually press the submit to admins for review button. Unless it's not there?

**Comment by jaskwong** [comment]
1. After submitting a node, it requires a page reload for the submitter to be able to add a comment (ie. submitter can't add node and then add comment, they need to add node, refresh page, add comment)

2 1 Public users can see the green outline (not sure if this is intended) for an accepted node

2.2 Public users can see the review section and comments from the admin for submitted nodes with a green outline

3. If the admin accepts a node, the outline remains green, if they then click on the node and publish it, the outline goes away and issues (2.1 and 2.2) go away (not sure if this is a bug or just works as designed)

**Comment by seanRong** [comment]
@jaskwong 
1 is intentional as it's not meant to be a comment that goes with the submission, rather it's a comment that explains why a node was rejected/accepted. 

Will work on the green line stuff.

**Comment by wynnset** [comment]
Small addition to scope: If the user tries to reject a node without leaving a comment, confirm with them that they want to reject without a comment. E.g. "Are you sure you want to reject this submission without a comment?"

**Comment by narendrasss** [comment]
Closing this in favor of #864

**Pull request closed without merging**